torchvision/ops/stochastic_depth.py

```diff
@@ -0,0 +1,56 @@
+import torch
+from torch import nn, Tensor
+
+
+def stochastic_depth(input: Tensor, mode: str, p: float, training: bool = True) -> Tensor:
+    """
+    Implements the Stochastic Depth from `"Deep Networks with Stochastic Depth"
+    <https://arxiv.org/abs/1603.09382>`_ used for randomly dropping residual
+    branches of residual architectures.
+
+    Args:
+        input (Tensor[N, ...]): The input tensor or arbitrary dimensions with the first one
+                    being its batch i.e. a batch with ``N`` rows.
+        mode (str): ``"batch"`` or ``"row"``.
+                    ``"batch"`` randomly zeroes the entire input, ``"row"`` zeroes
+                    randomly selected rows from the batch.
+        p (float): probability of the input to be zeroed.
+        training: apply dropout if is ``True``. Default: ``True``
+
+    Returns:
+        Tensor[N, ...]: The randomly zeroed tensor.
+    """
+    if p < 0.0 or p > 1.0:
+        raise ValueError("drop probability has to be between 0 and 1, but got {}".format(p))
+    if not training or p == 0.0:
+        return input
+
+    survival_rate = 1.0 - p
+    if mode == "batch":
+        keep = torch.rand(size=(1, ), dtype=input.dtype, device=input.device) < survival_rate
+    elif mode == "row":
+        keep = torch.rand(size=(input.size(0),), dtype=input.dtype, device=input.device) < survival_rate
+        keep = keep[(None, ) * (input.ndim - 1)].T
+    else:
+        raise ValueError("mode has to be either 'batch' or 'row', but got {}".format(mode))
+    return input / survival_rate * keep
```

Comment:
The `training` param doesn't seem to be used. Should we only devide by `survival_rate` when it's True?

Reply:
We use it, check few lines up. And no the formula is correct.

Reply:
From what I can tell we only use it here:

```py
    if not training or p == 0.0:
        return input
```

which makes me wonder whether this parameter is needed? If it's intended, I would suggest to update the docstring to:
> training: apply dropout if is ``True``. If ``False``, this transform is a no-op. Default: ``True``

Reply:
I'm following the same convention as in PyTorch's Dropout:
https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/dropout.py#L58

Reply:
Oh right, I forgot `self.training` gets set automatically. I'm wondering whether the current docstring is accurate though:

> training: apply dropout if is ``True``. Default: ``True``

Should `dropout` be something else?